### ISSUE TITLE
Feature Request: Allow login for people who want to edit/fix stations

### ISSUE THREAD

**Issue by ghost** [segler-alex/radiobrowser#121]
We need a way to maintain this database as a community and not allowing us to edit in any way is making the database more and more of a mess every day. We need a way to be able to edit by passing some kind of tests like entering a captcha correctly and providing login information.

**Comment by segler-alex** [comment]
this would not fix the problem. there is someone who hates this project so much, that he would do captchas jist to destroy it

**Comment by segler-alex** [comment]
login just makes things more complicated but does not fix the problem of bad people

**Comment by Riotous** [comment]
@segler-alex The only thing I can think of that would fix the problem of bad people is to have a trusted group of people to edit broken stations and process edit submissions, and I doubt we can do that unless volunteers step forward, and that would require an authentication system. 

I didn't realize somebody is actually attacking the project. I thought it was just general vandalism. Sorry to hear that.

**Comment by mhcpan** [comment]
@segler-alex why not enable the edit function for broken urls? #118

**Comment by mailinglists35** [comment]
>there is someone who hates this project so much

this can only be large streamer like apple radio or tunein, that feel their market is at risk because of a free solution. a small market player would not be affected by radiobrowser.

**Comment by wolfmanfp** [comment]
@mailinglists35 No, he's talking about a troll who constantly vandalised the station records to the point where Alex had to disable editing. It wasn't some big corporation at all.

Although I agree, there is a need for some kind of authorization.

**Comment by Riotous** [comment]
@wolfmanfp well, you never know about it not being a corporation. They certainly have an incentive to destroy an open radio list, which will drive users to their proprietary apps.

**Comment by h3298** [comment]
> @segler-alex this would not fix the problem. there is someone who hates this project so much, that he would do captchas jist to destroy it

> @mailinglists35 only a protection like wikipedia does for articles can be effective. this means lots of volunteers and an authentication system

If you combine voting with CAPTCHA and registration, a malicious edit could be reversed when flagged by a certain number of registered users.  Then volunteer editors would not be needed.  The comment platform "GetCivil" worked in a similar fashion.  (The company only closed because it did not make a profit.)

https://medium.com/@aja_15265/saying-goodbye-to-civil-comments-41859d3a2b1d

**Comment by wolfmanfp** [comment]
@Riotous I'm not speculating, that person is a troll living in Romania who has been vandalizing the entries, and personally attacking Alex for almost a year now. When editing was disabled, he created a GitHub account, which he deleted since. He claimed that he has much time on his hands, and he's still creating fake entries.
You can find remnants of this conversation here (his comments has been deleted with the account): https://github.com/segler-alex/radiobrowser/issues/93

**Comment by davidhedlund** [comment]
This issue is a duplication of https://github.com/segler-alex/radiobrowser/issues/99

**Comment by davidhedlund** [comment]
@segler-alex Any progress to model a new strategy?

**Comment by deepu2** [comment]
Someone added a false language tag to a station and I can't fix it.

**Comment by alexwender1** [comment]
We are here to help @segler-alex.

There are accumulating several reports of problems (incorrect logo, transmission link exchanged, incomplete name...); and I believe we can help (responding to those who forward complaints ... forwarding the modifications for you and/or accessing the database and correcting the information).

I don't want to see this project abandoned.

**Comment by wolfmanfp** [comment]
I totally agree with you. However, I think that @segler-alex won't enable editing until the new Rust-based API is ready.
Please, don't give this project up because of some idiot with no life. Just tell us, what needs to be done to finish the API, and we will help.

**Comment by dest4** [comment]
Allowing edits of broken URLs seems a good idea. #118 
Whitelisting a few IPs of "trusted" contributors could be a quick fix before the Rust API is ready. In this scenario I would volunteer to be one of them.
Proof the contributor is legit could be a request/commit from an old and populated Github account for example.

I built Adblock Radio on top of the Radio Browser API and would love to keep it that way.

**Comment by davidhedlund** [comment]
> Allowing edits of broken URLs seems a good idea. #118
> Whitelisting a few IPs of "trusted" contributors could be a quick fix before the Rust API is ready. In this scenario I would volunteer to be one of them.
> Proof the contributor is legit could be a request/commit from an old and populated Github account for example.
> 
> I built Adblock Radio on top of the Radio Browser API and would love to keep it that way.

Moderators could maintain radio stations for genres that they listen to themselves. For example, I could continue to maintain the [psytrance](http://www.radio-browser.info/gui/#!/bytag/psytrance) genre that I initially expanded from a few entries to 40 entries.

**Comment by davidhedlund** [comment]
Reddit is a site that has subreddit moderators (example: [https://www.reddit.com/r/batteries/about/moderators/](r/batteries)). Genre moderators for radio-browsers.info can't go wrong.

**Comment by davidhedlund** [comment]
https://github.com/segler-alex/radiobrowser-api/issues/54

**Comment by alexwender1** [comment]
It is better to divide by language; and users organize themselves, subdividing by country.

Music styles are infinite... And it seems superficial to me this form of division.

**Comment by kousu** [comment]
I don't think you need explicit authorization or moderators. I think you can get by with behavioural measurements e.g. [bayesian filtering](https://en.wikipedia.org/wiki/Naive_Bayes_spam_filtering) and [rate measurement](https://www.fail2ban.org/wiki/index.php/Main_Page), or [algorithmic rate limiting](http://www.hashcash.org/). Reduce the rate that vandalism can be added --- by detecting it statistically --- without reducing the rate that good edits are made. Good edits typically happen at a much lower rate than vandalism edits, which means hard-limiting everyone to a slow rate will deal with most of the problem.

Adding accounts means adding a whole human political element to the system. Maybe that's an issue that will eventually need addressing --- maybe there is no system that isn't political --- but it shifts the focus of Radio-Browser.info.

I would support having accounts with a higher edit rate for people that want to volunteer to go through and clean up stations in batches.

**Comment by alexwender1** [comment]
https://github.com/segler-alex/radiobrowser/issues/154
